# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build cairo-build harfbuzz-old-build"

PKG_CONFIG_PATH=/fb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.61; am-1.10; lt-1.5.24)
libtoolize -c -f; aclocal -I . -I m4; autoconf; autoheader; automake -a -c
rm -rf pango/opentype
PKG_CONFIG_PATH=/fb/lib/pkgconfig ./configure --with-system-harfbuzz --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/bin
install pango/.libs/pango-querymodules $DESTDIR/fb/bin
install pango-view/.libs/pangoft2-view $DESTDIR/fb/bin/pango-view
install -d $DESTDIR/fb/etc/pango
install -d $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-attributes.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-break.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-context.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-coverage.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-engine.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-enum-types.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-features.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-font.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-fontmap.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-fontset.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-glyph-item.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-glyph.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-gravity.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-item.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-language.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-layout.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-matrix.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-renderer.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-script.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-tabs.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-types.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pango-utils.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pangocairo.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pangofc-decoder.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pangofc-font.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pangofc-fontmap.h $DESTDIR/fb/include/pango-1.0/pango
install -m 644 pango/pangoft2.h $DESTDIR/fb/include/pango-1.0/pango
install -d $DESTDIR/fb/lib
install pango/.libs/libpango-1.0.so.0.* $DESTDIR/fb/lib/libpango-1.0.so.0
ln -sf libpango-1.0.so.0 $DESTDIR/fb/lib/libpango-1.0.so
install pango/.libs/libpangocairo-1.0.so.0.* $DESTDIR/fb/lib/libpangocairo-1.0.so.0
ln -sf libpangocairo-1.0.so.0 $DESTDIR/fb/lib/libpangocairo-1.0.so
install pango/.libs/libpangoft2-1.0.so.0.* $DESTDIR/fb/lib/libpangoft2-1.0.so.0
ln -sf libpangoft2-1.0.so.0 $DESTDIR/fb/lib/libpangoft2-1.0.so
install -d $DESTDIR/fb/lib/pango/1.6.0/modules
install modules/basic/.libs/pango-basic-fc.so $DESTDIR/fb/lib/pango/1.6.0/modules
install -d $DESTDIR/fb/lib/pkgconfig
install -m 644 pango.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pangocairo.pc $DESTDIR/fb/lib/pkgconfig
install -m 644 pangoft2.pc $DESTDIR/fb/lib/pkgconfig
LD_LIBRARY_PATH=$DESTDIR/fb/lib $DESTDIR/fb/bin/pango-querymodules > $DESTDIR/fb/etc/pango/pango.modules

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/pango-build.pc << EOF
Name: pango
Version: 1.20.5
Description: A library for layout and rendering of text
Requires: $REQUIRES

devel=\\
/fb/include/pango-1.0/pango/pango.h \\
/fb/include/pango-1.0/pango/pango-attributes.h \\
/fb/include/pango-1.0/pango/pango-break.h \\
/fb/include/pango-1.0/pango/pango-context.h \\
/fb/include/pango-1.0/pango/pango-coverage.h \\
/fb/include/pango-1.0/pango/pango-engine.h \\
/fb/include/pango-1.0/pango/pango-enum-types.h \\
/fb/include/pango-1.0/pango/pango-features.h \\
/fb/include/pango-1.0/pango/pango-font.h \\
/fb/include/pango-1.0/pango/pango-fontmap.h \\
/fb/include/pango-1.0/pango/pango-fontset.h \\
/fb/include/pango-1.0/pango/pango-glyph-item.h \\
/fb/include/pango-1.0/pango/pango-glyph.h \\
/fb/include/pango-1.0/pango/pango-gravity.h \\
/fb/include/pango-1.0/pango/pango-item.h \\
/fb/include/pango-1.0/pango/pango-language.h \\
/fb/include/pango-1.0/pango/pango-layout.h \\
/fb/include/pango-1.0/pango/pango-matrix.h \\
/fb/include/pango-1.0/pango/pango-renderer.h \\
/fb/include/pango-1.0/pango/pango-script.h \\
/fb/include/pango-1.0/pango/pango-tabs.h \\
/fb/include/pango-1.0/pango/pango-types.h \\
/fb/include/pango-1.0/pango/pango-utils.h \\
/fb/include/pango-1.0/pango/pangocairo.h \\
/fb/include/pango-1.0/pango/pangofc-decoder.h \\
/fb/include/pango-1.0/pango/pangofc-font.h \\
/fb/include/pango-1.0/pango/pangofc-fontmap.h \\
/fb/include/pango-1.0/pango/pangoft2.h \\
/fb/lib/libpango-1.0.so \\
/fb/lib/libpangocairo-1.0.so \\
/fb/lib/libpangoft2-1.0.so \\
/fb/lib/pkgconfig/pango.pc \\
/fb/lib/pkgconfig/pangocairo.pc \\
/fb/lib/pkgconfig/pangoft2.pc

exec=\\
/fb/bin/pango-querymodules \\
/fb/bin/pango-view \\
/fb/etc/pango/pango.modules \\
/fb/lib/libpango-1.0.so.0 \\
/fb/lib/libpangocairo-1.0.so.0 \\
/fb/lib/libpangoft2-1.0.so.0 \\
/fb/lib/pango/1.6.0/modules/pango-basic-fc.so
EOF

exit
--- pango-1.20.5.orig/configure.in
+++ pango-1.20.5/configure.in
@@ -183,6 +183,10 @@
                               [disable all source autogeneration rules])],,
               [enable_rebuilds=yes])
 
+AC_ARG_WITH(system-harfbuzz,
+   AC_HELP_STRING([--with-system-harfbuzz],
+                  [use system harfbuzz library @<:@default=no@:>@]))
+
 AM_CONDITIONAL(CROSS_COMPILING, [test $cross_compiling = yes])
 
 # define a MAINT-like variable REBUILD which is set if Perl
@@ -235,7 +239,17 @@
   # By freetype, we really always mean freetype+fontconfig
   FREETYPE_CFLAGS="$FREETYPE_CFLAGS $FONTCONFIG_CFLAGS"
   FREETYPE_LIBS="$FREETYPE_LIBS $FONTCONFIG_LIBS"
+
+  if test "x$with_system_harfbuzz" = "xyes"; then
+    PKG_CHECK_MODULES(OPENTYPE, harfbuzz-old)
+  else
+    OPENTYPE_LIBS="opentype/libharfbuzz-1.la"
+    OPENTYPE_DEPS="$OPENTYPE_LIBS"
+    OPENTYPE_SUBDIR=opentype
+  fi
 fi
+AC_SUBST(OPENTYPE_DEPS)
+AC_SUBST(OPENTYPE_SUBDIR)
 AM_CONDITIONAL(HAVE_FREETYPE, $have_freetype)
 
 #
@@ -913,7 +927,6 @@
 Makefile 
 pango/Makefile
 pango/mini-fribidi/Makefile
-pango/opentype/Makefile
 pango/pango.rc
 pango/pangoft2.rc
 pango/pangowin32.rc
@@ -948,6 +961,10 @@
 pango-zip.sh
 ])
 
+if test "x$with_system_harfbuzz" != "xyes"; then
+  AC_CONFIG_FILES([pango/opentype/Makefile])
+fi
+
 AC_CONFIG_FILES([tests/runtests.sh],
 	        [chmod +x tests/runtests.sh])
 
--- pango-1.20.5.orig/m4/gtk-doc.m4
+++ pango-1.20.5/m4/gtk-doc.m4
@@ -0,0 +1,39 @@
+dnl -*- mode: autoconf -*-
+
+# serial 1
+
+dnl Usage:
+dnl   GTK_DOC_CHECK([minimum-gtk-doc-version])
+AC_DEFUN([GTK_DOC_CHECK],
+[
+  AC_BEFORE([AC_PROG_LIBTOOL],[$0])dnl setup libtool first
+  AC_BEFORE([AM_PROG_LIBTOOL],[$0])dnl setup libtool first
+  dnl for overriding the documentation installation directory
+  AC_ARG_WITH([html-dir],
+    AS_HELP_STRING([--with-html-dir=PATH], [path to installed docs]),,
+    [with_html_dir='${datadir}/gtk-doc/html'])
+  HTML_DIR="$with_html_dir"
+  AC_SUBST([HTML_DIR])
+
+  dnl enable/disable documentation building
+  AC_ARG_ENABLE([gtk-doc],
+    AS_HELP_STRING([--enable-gtk-doc],
+                   [use gtk-doc to build documentation [[default=no]]]),,
+    [enable_gtk_doc=no])
+
+  if test x$enable_gtk_doc = xyes; then
+    ifelse([$1],[],
+      [PKG_CHECK_EXISTS([gtk-doc],,
+                        AC_MSG_ERROR([gtk-doc not installed and --enable-gtk-doc requested]))],
+      [PKG_CHECK_EXISTS([gtk-doc >= $1],,
+                        AC_MSG_ERROR([You need to have gtk-doc >= $1 installed to build gtk-doc]))])
+  fi
+
+  AC_MSG_CHECKING([whether to build gtk-doc documentation])
+  AC_MSG_RESULT($enable_gtk_doc)
+
+  AC_PATH_PROGS(GTKDOC_CHECK,gtkdoc-check,)
+
+  AM_CONDITIONAL([ENABLE_GTK_DOC], [test x$enable_gtk_doc = xyes])
+  AM_CONDITIONAL([GTK_DOC_USE_LIBTOOL], [test -n "$LIBTOOL"])
+])
--- pango-1.20.5.orig/pango/Makefile.am
+++ pango-1.20.5/pango/Makefile.am
@@ -4,15 +4,11 @@
 # which are more like other make's VPATH.
 GPATH = $(srcdir)
 
-if HAVE_FREETYPE
-OPENTYPE_SUBDIR=opentype
-endif
-
 FRIBIDI_SUBDIR=mini-fribidi
 
 SUBDIRS = $(OPENTYPE_SUBDIR) $(FRIBIDI_SUBDIR)
 
-DIST_SUBDIRS = mini-fribidi opentype
+DIST_SUBDIRS = mini-fribidi
 
 INCLUDES =						\
 	-DG_LOG_DOMAIN=\"Pango\"			\
@@ -189,16 +185,17 @@
 lib_LTLIBRARIES += libpangoft2-1.0.la
 endif
 
+libpangoft2_1_0_la_CFLAGS = $(OPENTYPE_CFLAGS)
 libpangoft2_1_0_la_LDFLAGS = $(LIBRARY_LIBTOOL_OPTIONS)
 libpangoft2_1_0_la_LIBADD =			\
-	opentype/libharfbuzz-1.la		\
+	$(OPENTYPE_LIBS)			\
 	libpango-$(PANGO_API_VERSION).la	\
 	$(INCLUDED_FC_MODULES)			\
 	$(GLIB_LIBS)				\
 	$(FREETYPE_LIBS)			\
 	$(libm)
 libpangoft2_1_0_la_DEPENDENCIES =		\
-	opentype/libharfbuzz-1.la		\
+	$(OPENTYPE_DEPS)			\
 	libpango-$(PANGO_API_VERSION).la	\
 	$(INCLUDED_FC_MODULES)
 libpangoft2_1_0_la_SOURCES =   	\
